### DISCUSSION TITLE
Windows OS - no admin rights

### DISCUSSION THREAD

**Discussion by syed-shahjahan** [coreybutler/nvm-windows#598] [General] [answered]
This is an amazing tool, switching between different nodeJS versions without multiple installations.

The problem I am facing with Windows OS is Admin rights. Generally, we get a laptop from companies with having **no admin rights** to make changes. So that unable to use this tool.

Hope the same problem is faced by most of the developers working in IT companies.

Really appreciated it if we get an alternative solution for these admin rights problems.

**Comment by coreybutler** [comment, marked as answer]
Unfortunately, there is no workaround for this issue because Windows itself requires administrative rights in order to create symlinks. You also need admin rights in order to install this utility, since it writes to the registry (like most Windows installations).

**Reply by BurtHarris** [reply]
@coreybutler:   recent builds of Windows 10 have made this answer incorrect.  It no longer requires administrative rights, nor developer mode to create symlinks in general.   The restriction currently being encountered is that file operations in "C:\Program Files" do require admin rights.   This can be addressed simply now by setting both NVM_HOME _and_ NVM_SYMLINK to locations in the user's profile.   

If the installer were changed to not put NVM_SYMLINK in C:\Program Files by default, I think the problems of requiring admin rights would now disappear.

P.S. Ideally these locations would be in %USERPROFILE/AppData/Local, not in the Roaming part of the profile.

**Reply by coreybutler** [reply]
@BurtHarris - From my research, I still see a requirement for developer mode. If you have a link to a resource stating otherwise, I'd be happy to take a look.

Many organizations will not allow installations outside of `C:\Program Files`. The default value could be changed in the installer, but forcing a user profile install requires the app to be installed for each user instead of a system-wide basis. For individual developers that is fine, but that's a problem on shared servers where multiple people login. In other words, the multiuser install cannot be ignored.

**Comment by mmunte-impeo** [comment]
This is actually not very accurate. Windows 10 allows to create symlinks if
- the user owns the target folder
- development mode is enabled

There has been a change of the runelevated.cmd in BIN, t try to execute the commands needed by nvm-windows without UAC.
But this updated version seems to have never been released. In the GIT repo, version is 1.1.8, but under releases I can only find 1.1.7

So, 
- I took my 1.1.7 installation, found the folder where nvm is installed and replaced runelevated.cmd with the newest version
- enabled windows 10 development mode
- changed the env var NVM_SYMLINK to point to c:\nodejs (away from program files since this folder is UAC protected)

And with the above changes I can confirm that nvm-windows runs without a UAC prompt. This is great as my dev team needs it badly. 

@coreybutler any reason for not releasing 1.1.8?

THANKS A LOT FOR THIS UTILITY
